### PR TITLE
Make macsec container to build with bookworm

### PR DESCRIPTION
**Pull request by wumiaont**
#### Why I did it
We Nokia took the task to make macsec container to work with bookworm. 

This PR requires another PR for sonic-wpa-supplicant(https://github.com/sonic-net/sonic-wpa-supplicant/pull/80) to be in first. Otherwise sonic build will fail for bookworm wpasupplicant debian package.

#### How to verify it
After build and boot up system with new image, macsec container should run with bookworm. Corresponding mgmt macsec suite will work here. 
#### Tested branch (Please provide the tested image version)
Tested with latest master of sonic-buildimage. Built vs image and loaded into vs testing environment. Code pass with macsec test suites.

#### Description for the changelog
Make macsec container to build with bookworm

### CODE REVIEW

**Comment by wumiaont** [comment]
Can some of you help to review https://github.com/sonic-net/sonic-wpa-supplicant/pull/80. The failures are because of that.

**Comment by judyjoseph** [comment]
> Can some of you help to review [sonic-net/sonic-wpa-supplicant#80](https://github.com/sonic-net/sonic-wpa-supplicant/pull/80). The failures are because of that.

Will wait till this PR is merged

File: rules/docker-macsec.mk
Comment (saiarcot895) [reply rules/docker-macsec.mk:9]: Yes, if macsec has a dependency on swss, please modify the container to use docker-swss-layer-bookworm instead.

Comment (wumiaont) [reply rules/docker-macsec.mk]: Made changes as suggestion.

**Comment by wumiaont** [comment]
@saiarcot895 @k-v1 help to review it again? Thx

**Comment by wumiaont** [comment]
@saiarcot895 @k-v1 Semgrep did not run. Anyway you can help to trigger this check?

**Comment by wumiaont** [comment]
@saiarcot895 @k-v1 @xumia Help to review and approve the PR? Thanks.

**Comment by wumiaont** [comment]
@qiluo-msft help to review the PR? Seems needs reviewers with write access to approve.

**Review by xumia — APPROVED** [review]
LGTM

**Pull request merged**
